src/network/networkd-dhcp6.c

```diff
@@ -1043,6 +1043,39 @@ static bool dhcp6_link_has_dhcpv6_prefix(Link *link) {
         return false;
 }
 
+static int dhcp6_assign_delegated_prefix_address_handler(sd_netlink *rtnl, sd_netlink_message *m, Link *link) {
+          int r;
+
+          assert(rtnl);
+          assert(m);
+          assert(link);
+          assert(link->ifname);
+          assert(link->dhcp6_delegation_prefix_address_messages > 0);
+          assert(IN_SET(link->state, LINK_STATE_CONFIGURING,
+                 LINK_STATE_FAILED, LINK_STATE_LINGER));
+
+          link->dhcp6_delegation_prefix_address_messages--;
+
+          if (IN_SET(link->state, LINK_STATE_FAILED, LINK_STATE_LINGER))
+                  return 1;
+
+          r = sd_netlink_message_get_errno(m);
+          if (r < 0 && r != -EEXIST) {
+                  log_link_message_warning_errno(link, m, r, "Could not set DHCP6 delegated prefix address");
+                  link_enter_failed(link);
+                  return 1;
+          } else if (r >= 0)
+                  (void) manager_rtnl_process_address(rtnl, m, link->manager);
+
+          if (link->dhcp6_delegation_prefix_address_messages == 0) {
+                  log_link_debug(link, "DHCP6 deletaged prefix addresses set");
+                  link->dhcp6_delegation_prefix_address_configured = true;
+                  link_check_ready(link);
+          }
+
+          return 1;
+  }
```

Comment:
This is exactly the same as `address_handler`, but uses `dhcp6_delegation_prefix_address_messages` for counting.
I don't see how this would fix the issue.

Reply:
How you think we can fix this ?